src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -671,14 +672,15 @@ public override TDeclarationNode UpdateDeclarationAccessibility<TDeclarationNode
 
         private static SyntaxTokenList UpdateDeclarationAccessibility(SyntaxTokenList modifiersList, Accessibility newAccessibility, CodeGenerationOptions options)
         {
-            var newModifierTokens = new List<SyntaxToken>();
+            var newModifierTokens = ArrayBuilder<SyntaxToken>.GetInstance();
             CSharpCodeGenerationHelpers.AddAccessibilityModifiers(newAccessibility, newModifierTokens, options, Accessibility.NotApplicable);
             if (newModifierTokens.Count == 0)
             {
                 return modifiersList;
             }
 
-            return GetUpdatedDeclarationAccessibilityModifiers(newModifierTokens, modifiersList, (SyntaxToken modifier) => SyntaxFacts.IsAccessibilityModifier(modifier.Kind()))
+            // TODO: check ToList usage
+            return GetUpdatedDeclarationAccessibilityModifiers(newModifierTokens.ToList(), modifiersList, (SyntaxToken modifier) => SyntaxFacts.IsAccessibilityModifier(modifier.Kind()))
                 .ToSyntaxTokenList();
```

Comment:
use ToSyntaxTokenListAndFree?

Reply:
I would like to move this API to ArrayBuilder but it ripples out to VB as well. I'd like to say this is "good enough" for this PR (just need it to not leak and not be blatantly wasteful) and create a tracking issue to move to pooled collections across CodeGenerationService. What do you think?

Reply:
works for me!

Reply:
Makes sense not fixing everywhere, but ToListAndFree() still exists, right?

Reply:
I don't see ToListAndFree on ArrayBuilder. Maybe I'm not looking in the right place?

Reply:
Tracking in #34960